### PR TITLE
be a bit more carefull with payment requests

### PR DESCRIPTION
**Pull request by marcelhollerbach**
a request to pay will fail if the given list of paymentjobs does not fit
with the workers which have submitted a answer.

### CODE REVIEW

**Comment by marcelhollerbach** [comment]
@LeanderK

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/PlatformManager.java [hunk 1]
```diff
@@ -249,11 +249,23 @@ public String identifyWorker(String name, Map<String, String[]> params) throws U
...
+     * @return a future object which indicates if the payment was successful or not
+     * @throws IllegalWorkerSetException if this exception is thrown NO payment requests are given to the platform
+     *
      */
-    public CompletableFuture<Boolean> payExperiment(String name, Experiment experiment, List<PaymentJob> paymentJobs) throws TaskOperationException {
+    public CompletableFuture<Boolean> payExperiment(String name, Experiment experiment, List<PaymentJob> paymentJobs) throws TaskOperationException, IllegalWorkerSetException {
         TaskRecord record = tasksOps.getTask(name, experiment.getId()).
                 orElseThrow(() -> new TaskOperationException("Experiment was never published"));
+        List<WorkerRecord> workerRecords = workerOps.getWorkerWithWork(experiment.getId(), name);
+        List<String> given, should;
```
Comment (simonkorz) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/PlatformManager.java]: bad style

**Comment by LeanderK** [comment]
@marcelhollerbach push

**Comment by marcelhollerbach** [comment]
yo, is it possible that the tests are frozen ?

**Comment by kelunik** [comment]
It's running.

**Comment by marcelhollerbach** [comment]
Heureka - well the button

**Pull request merged**